salt/syspaths.py

```diff
@@ -90,11 +91,40 @@
 CLOUD_DIR = os.path.join(INSTALL_DIR, "cloud")
 BOOTSTRAP = os.path.join(CLOUD_DIR, "deploy", "bootstrap-salt.sh")
 
+
+def _get_windows_root_dir():
+    # Try to get the root directory location from the registry
+    # This key will be created by the NullSoft installer
+    # If salt is currently installed in C:\salt and the user performs an
+    # upgrade, then this key will be set to C:\salt
+    # TODO: Probably need to lock down this key in salt.utils.verify.py
+    # TODO: and in the NullSoft installer code
+    root_dir = salt.utils.win_reg.read_value(
+        hive="HKLM", key="SOFTWARE\\Salt Project\\salt", vname="root_dir"
+    )
+    if root_dir["success"]:
+        return root_dir
+    else:
+        # If this key does not exist, then salt was not installed using the
+        # installer. Could be pip or setup.py.
+        log.warning(root_dir["comment"])
```

Comment:
A better log message? And a lower log level?

Reply:
That comment is the following:
```
Cannot find key: HKLM\SOFTWARE\Salt Project\salt
```